nexus/types/src/external_api/params.rs

```diff
@@ -1755,18 +1829,71 @@ pub struct Address {
     pub vlan_id: Option<u16>,
 }
 
+/// An address to be added or removed from an interface
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct AddressAddRemove {
+    /// The name of the interface
+    pub interface: Name,
+
+    /// The address lot this address is drawn from.
+    pub address_lot: NameOrId,
+
+    /// The address and prefix length of this address.
+    pub address: IpNet,
+
+    /// Optional VLAN ID for this address
+    pub vlan_id: Option<u16>,
+}
+
 /// Select a port settings object by an optional name or id.
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
 pub struct SwitchPortSettingsSelector {
-    /// An optional name or id to use when selecting port settings.
-    pub port_settings: Option<NameOrId>,
+    /// An optional name or id to use when selecting a switch port configuration.
+    pub configuration: Option<NameOrId>,
 }
 
 /// Select a port settings info object by name or id.
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
 pub struct SwitchPortSettingsInfoSelector {
-    /// A name or id to use when selecting switch port settings info objects.
-    pub port: NameOrId,
+    /// A name or id to use when selecting a switch port configuration.
+    pub configuration: NameOrId,
+}
+
+/// Select a port settings info object by name or id and peer by address.
```

Comment:
"port settings info object" caught my eye because it is identical to line 1855 above, but should probably be different.